code/modules/mob/living/carbon/xenomorph/castes/warlock/abilities_warlock.dm

```diff
@@ -0,0 +1,416 @@
+		if(!uncrossing)
+			proj.uncross_scheduled += src
+		return FALSE
+	if (uncrossing)
+		return FALSE
+	reflect_projectile(proj)
+	return FALSE
+
+///Reflects projectiles based on their relative incoming angle
+/obj/effect/xeno/shield/proc/reflect_projectile(obj/projectile/proj)
+	playsound(loc, 'sound/effects/portal.ogg', 20)
+	var/new_range = proj.proj_max_range - proj.distance_travelled
+
+	if(new_range <= 0)
+		proj.ammo.do_at_max_range(proj)
+		qdel(proj)
+		return
+	var/perpendicular_angle = Get_Angle(src, get_step(src, dir)) //the angle src is facing
+	var/new_angle = (perpendicular_angle + (perpendicular_angle - proj.dir_angle - 180))
+	if(new_angle < 0)
+		new_angle += 360
+	else if(new_angle > 360)
+		new_angle -= 360
+	proj.firer = src
+	proj.fire_at(shooter = src, source = src, range = new_range, angle = new_angle, recursivity = TRUE) //loc_override = loc //note investigate hitscan issue
+
+// ***************************************
+// *********** psychic crush
+// ***************************************
+/datum/action/xeno_action/activable/psy_crush
+	name = "psychic Crush"
+	action_icon_state = "psy_crush"
+	mechanics_text = "Channel an expanding AOE crush effect, activating it again pre-maturely crushes enemies over an area. The longer it is channeled, the larger area it will affect, but will consume more plasma."
+	ability_name = "psychic crush"
+	plasma_cost = 35
+	cooldown_timer = 8 SECONDS
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_PSYCHIC_CRUSH,
+	)
+	///The number of times we can expand our effect radius. Effectively a max radius
+	var/max_interations = 5
+	///How many times we have expanded our effect radius
+	var/current_iterations = 0
+	///timer hash for the timer we use when charging up
+	var/channel_loop_timer
+	///List of turfs in the effect radius
+	var/list/target_turfs = list()
+	///list of effects used to visualise area of effect
+	var/list/effect_list = list()
+	/// A list of all things that had a fliter applied
+	var/list/filters_applied = list()
+	///max range at which we can cast out ability
+	var/ability_range = 9
+
+/datum/action/xeno_action/activable/psy_crush/use_ability(atom/target)
+	if(channel_loop_timer)
+		if(length(target_turfs)) //it shouldn't be possible to do this without any turfs, but just in case
+			crush(target_turfs[1])
+		return
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	if(xeno_owner.selected_ability != src)
+		action_activate()
+		return
+	if(owner.do_actions || !target || !can_use_action(TRUE) || !check_distance(target))
+		return fail_activate()
+	if(!do_after(owner, 0.5 SECONDS, TRUE, owner, BUSY_ICON_DANGER, extra_checks = CALLBACK(src, .proc/can_use_action, FALSE, XACT_USE_BUSY)))
+		return fail_activate()
+	owner.visible_message(span_xenowarning("\The [owner] starts channeling their psychic might!"), \
+		span_xenowarning("We start channeling our psychic might!"))
```

Comment:
is this visible message really needed

Reply:
Agreed, although I need a visual indictor they're charging. Givparticleeditor.